src/signature.rs

```diff
@@ -79,12 +105,10 @@ pub fn sign(secret: &Secret, message: &Message) -> Result<Signature, Error> {
 	let sec: &SecretKey = unsafe { mem::transmute(secret) };
 	let s = try!(context.sign_recoverable(&try!(SecpMessage::from_slice(message.deref())), sec));
 	let (rec_id, data) = s.serialize_compact(context);
-	let mut signature = Signature::default();
-	signature.r.copy_from_slice(&data[0..32]);
-	// no need to check if s is low, it alawys is
-	signature.s.copy_from_slice(&data[32..64]);
-	signature.v = rec_id.to_i32() as u8;
-	Ok(signature)
+	let mut data_arr = [0; 65];
+	(&mut data_arr[0..64]).copy_from_slice(&data[0..64]);
```

Comment:
it can be just `data_arr[0..64].copy_from_slice(&data[0..64]);` https://is.gd/drDiDM